pkg/types/validation/installconfig.go

```diff
@@ -268,3 +271,19 @@ func validateProxy(p *types.Proxy, fldPath *field.Path) field.ErrorList {
 
 	return allErrs
 }
+
+func validateReleaseImageSources(sources []types.ReleaseImageSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for _, source := range sources {
+		ref, err := dockerref.ParseNamed(source.Repository)
```

Comment:
This would be better with `ParseNormalizedNamed`.  `ParseNamed` is `ParseNormalizedNamed`+ a check that the input is in one specified form (i.e. you must say `docker.io/openshift/foo` instead of `openshift/foo`).

Sure, assuming the content is not going to be hosted on `docker.io`, this does not matter all that much for _this_ code.

It’s rare that you _need_ to forbid the alternative forms, but copy&pasting code across projects could cause the `ParseNamed` calls to spread to other user-facing code that _should_ accept other forms and use `ParseNormalizednamed`.

OTOH the tests do explicitly test for rejecting the implied-`docker.io` forms, so if this was intentional, fine.

---

BTW if this is going to be long-term supporting Windows/containerd / whole-registry mirrors, note that `Parse{Normalized,}Named` does not accept registry hostnames (without repository specification). Everything here says “repository”, so it’s OK right now.

Reply:
So I think we do want to make sure people use complete names. no implicit names are allowed.

> BTW if this is going to be long-term supporting Windows/containerd / whole-registry mirrors, note that Parse{Normalized,}Named does not accept registry hostnames (without repository specification)

can you expand a little with examples of how this will break for windows...?

Reply:
`reference.ParseNamed("a.example.com")` interprets the input as `docker.io/library/a.example.com` and rejects it with `"repository name must be canonical"`.

Reply:
> `reference.ParseNamed("a.example.com")` interprets the input as `docker.io/library/a.example.com` and rejects it with `"repository name must be canonical"`.

I think it is rejecting it correctly right?

Reply:
Yes, “correctly” in the sense that the code wanted a repository in canonical form, the input was not a repository in canonical form.

Just pointing out that eventually we might need to support hostnames as well — but the validation can certainly be relaxed (or a separate, differently-validated field, can be added) later when/if that arises.

Reply:
… FWIW the context is that CRI-O does not currently run on Windows, and containerd’s mirroring configuration is only at the registry (= host:port) resolution.  (Of course, this particular bit of configuration would be one of the smaller things that need changing, the whole mirroring process would need adapting for the idea that the mirror must mirror all of (the relevant parts of) the registry, not just a single repository.)